lib/sequence.js

```diff
@@ -39,26 +38,41 @@ Sequence.prototype._run = function (fromIndex) {
 
 		var result = this.tests[i].run();
 
+		// if a Promise returned, we don't need to check for Promises after this test
+		// so we can just use Promise.each() on the rest of the tests
```

Comment:
`is-promise` only checks for `then()` methods though, yet the promise chain below assumes `result` is actually a Bluebird promise. That's probably guaranteed given the rest of the code but feels like a refactoring hazard.

Why not always use `Promise.resolve(result)` and the chain below? Or is that too expensive performance wise? I couldn't immediately find a "is this a Bluebird promise" function in their docs, but maybe `result instanceof Promise` will work for us?